src/mlpack/methods/ann/gan_impl.hpp

```diff
@@ -0,0 +1,392 @@
+
+  // Get the gradients of the Discriminator.
+  std::cout << "Calling Discriminator Gradient:" << std::endl;
+  discriminator.Gradient(discriminator.parameter, i, gradientDiscriminator,
+      batchSize);
+  //std::cout << "Updated Discriminator Gradient:" << std::endl << gradientDiscriminator << std::endl;
+
+  noise.imbue( [&]() { return noiseFunction();} );
+  //std::cout << "Noise:" << std::endl << noise << std::endl;
+  std::cout << "Calling Generator Forward:" << std::endl;
+  generator.Forward(std::move(noise));
+  //std::cout << "Discriminator Predictors:" << std::endl << discriminator.predictors << std::endl;
+  discriminator.predictors.col(numFunctions) = boost::apply_visitor(
+      outputParameterVisitor, generator.network.back());
+  //std::cout << "Updated Discriminator Predictors:" << std::endl << discriminator.predictors << std::endl;
+
+  discriminator.responses(numFunctions) = 0;
+  //std::cout << "Discriminator Responses:" << std::endl << discriminator.responses << std::endl;
+  //std::cout << "Noise Discriminator Gradient:" << std::endl << noiseGradientDiscriminator << std::endl;
+  std::cout << "Calling Discriminator Gradient:" << std::endl;
+  discriminator.Gradient(discriminator.parameter, numFunctions,
+      noiseGradientDiscriminator, batchSize);
+  //std::cout << "Updated Noise Discriminator Gradient:" << std::endl << noiseGradientDiscriminator << std::endl;
+  //std::cout << "Discriminator Gradient:" << std::endl << gradientDiscriminator << std::endl;
+  gradientDiscriminator += noiseGradientDiscriminator;
+  //std::cout << "Updated Discriminator Gradient:" << std::endl << gradientDiscriminator << std::endl;
+
+  if (currentBatch % generatorUpdateStep == 0 && preTrainSize == 0)
+  {
+    // Minimize -log(D(G(noise))).
+    // Pass the error from Discriminator to Generator.
+    std::cout << "Minimizing Error Objective:" << std::endl;
+    //std::cout << "Discriminator Responses:" << std::endl << discriminator.responses << std::endl;
+    discriminator.responses(numFunctions) = 1;
+    //std::cout << "Updated Discriminator Responses:" << std::endl << discriminator.responses << std::endl;
+    //std::cout << "Noise Discriminator Gradient:" << std::endl << noiseGradientDiscriminator << std::endl;
+    std::cout << "Calling Discriminator Gradient:" << std::endl;
+    discriminator.Gradient(discriminator.parameter, numFunctions,
+      noiseGradientDiscriminator, batchSize);
+    //std::cout << "Updated Noise Discriminator Gradient:" << std::endl << noiseGradientDiscriminator << std::endl;
+    generator.error = boost::apply_visitor(deltaVisitor,
+        discriminator.network[1]);
+    //std::cout << "Generator Error:" << std::endl << generator.error << std::endl;
+    //std::cout << "Noise:" << std::endl << noise << std::endl;
+    generator.currentInput = noise;
+    std::cout << "Calling Generator Backward:" << std::endl;
+    generator.Backward();
```

Comment:
``generator.Gradient(..)`` already calls the backward function, so I think this is unnecessary.